Library/Homebrew/utils.rb

```diff
@@ -337,6 +337,14 @@ def with_system_path
   ENV["PATH"] = old_path
 end
 
+def with_custom_locale(locale)
+  old_locale = ENV["LC_ALL"]
+  ENV["LC_ALL"] = locale
+  yield
+ensure
+  ENV["LC_ALL"] = old_locale
+end
```

Comment:
I lean 👎 on this for now.

I'll reconsider if it gets to the point where we're carrying around a silly amount of custom methods for relatively small cases, but I _personally_ find generic methods make handling a large codebase sometimes unpleasant, particularly around navigating & tying together specific elements efficiently.

But as said, if it gets to a point where this gets silly I'll accept the need. Hopefully that's a reasonable compromise.